core/src/core/transaction.rs

```diff
@@ -558,12 +533,68 @@ impl TransactionBody {
 	/// Validates all relevant parts of a transaction body. Checks the
 	/// excess value against the signature as well as range proofs for each
 	/// output.
-	pub fn validate(&self, with_reward: bool) -> Result<(), Error> {
+	pub fn validate(
+		&self,
+		with_reward: bool,
+		verifier: Arc<RwLock<VerifierCache>>,
+	) -> Result<(), Error> {
 		self.verify_weight(with_reward)?;
 		self.verify_sorted()?;
 		self.verify_cut_through()?;
-		self.verify_rangeproofs()?;
-		self.verify_kernel_signatures()?;
+
+		// TODO
+		// move verify_rangeproofs() and verify_kernel_signatures()
+		// back in here
+		// but we need something in util that will let us call with
+		// a vec of outputs and a cache
+		// given a vec of outputs filter into smaller vec based on cached values
+		// read cache for multiple outputs
+		// write cache for multiple outputs etc.
```

Comment:
Maybe have the verifier itself do that filtering, from a `Vec<output>` and `Vec<TxKernel>`?